sycl/include/CL/sycl/handler.hpp

```diff
@@ -1360,6 +1350,49 @@ class __SYCL_EXPORT handler {
 #endif
   }
 
+  /// Defines and invokes a SYCL kernel function for the specified nd_range.
+  ///
+  /// The SYCL kernel function is defined as a lambda function or a named
+  /// function object type and given an id for indexing in the indexing
+  /// space defined by range \p Range.
+  /// The parameter \p Redu contains the object creted by the reduction()
+  /// function and defines the type and operation used in the corresponding
+  /// argument of 'reducer' type passed to lambda/functor function.
+  template <typename KernelName = detail::auto_name, typename KernelType,
+            int Dims, typename Reduction>
+  void parallel_for(range<Dims> Range, Reduction Redu,
+                    _KERNELFUNCPARAM(KernelFunc)) {
+    shared_ptr_class<detail::queue_impl> QueueCopy = MQueue;
+
+    // Before running the kernels, check that device has enough local memory
+    // to hold local arrays required for the tree-reduction algorithm.
+    constexpr bool IsTreeReduction =
+        !Reduction::has_fast_reduce && !Reduction::has_fast_atomics;
+    size_t OneElemSize =
+        IsTreeReduction ? 0 : sizeof(typename Reduction::result_type);
+#ifdef SYCL_REDUCTION_NUM_EU_THREADS
+    uint32_t NumEUThreads = SYCL_REDUCTION_NUM_EU_THREADS;
+#else
+    uint32_t NumEUThreads = ONEAPI::detail::reduGetMaxNumEUThreads(MQueue);
+#endif
```

Comment:
I'd recommend replacing this with a term that isn't tied to Intel GPUs, that's more descriptive of how the variable is used.  Something like "NumConcurrentWorkGroups", maybe?

Reply:
It is unclear if that should be Groups or SubGroups. Would you agree on this?
uint32_t NumConcurrentThreads = ONEAPI::detail::reduGetMaxNumConcurrentThreads();

BTW, I forgot to remove "#ifdef SYCL_REDUCTION_NUM_EU_THREADS" code. It was for debugging, performance runs. 
I am going to either delete that code, OR to add underscores before the macro name and rename it to __SYCL_REDUCTION_NUM_CONCURRENT_THREADS

Reply:
I used the name suggested by you. 808bfa5122a6135f8e6feb45886cc0a59af18528